### DISCUSSION TITLE
Increase input query limit on public apps

### DISCUSSION THREAD

**Discussion by thosakwe** [Shopify/function-examples#475] [Feedback: General Improvements]
Public apps have to meet the needs of hundreds, if not thousands, of stores.

The input query limit of 30 makes sense for custom apps, where only 1 store is using the app, but prevents public apps from implementing as many features as their users would like.

For example, our app provides a wide array of conditions for merchants to create advanced automatic discounts. As we get close to the limit of 30, it's impossible to implement some popular requests.

Proposal: Increase the limit for public apps (or at least Built for Shopify apps) to 60.

**Comment by thosakwe** [comment]
One question that comes to mind: why is there a limit, especially since the query will only run once per Function invocation?

Because Functions are deterministic, I assume that their output can also be memoized.

I'm assuming it's performance related, but what would be the implications if the limit were as high as 100, or even unlimited?

**Reply by nickwesselman** [reply]
A higher GraphQL limit means more data to calculate and serialize to your Function, every time it's executed. And the more fields you have in the query, the less cacheable the execution.

**Comment by nickwesselman** [comment]
I understand the need for more flexibility for public apps, but there's also a higher responsibility for quality and performance on those apps (see also Built for Shopify). I do not see us making any special allowance for public apps on the Function platform limits.

We may look for ways to raise these for all apps in the future. Meanwhile I'd encourage you to explore creating more specialized function (discount) types, vs. a one-size-fits-all approach.

**Reply by thosakwe** [reply]
Thank you for taking the time to write this explanation.

**Comment by mitesh-avid-brio** [comment]
@thosakwe did you found something else ? like compression technique??

**Reply by thosakwe** [reply]
Please upvote and comment on this thread: https://github.com/Shopify/function-examples/discussions/566

**Reply by mitesh-avid-brio** [reply]
already did, have you found something else??
